syntax = "proto2";

package botCode;

message LightState {

  message AgentState {
    required int32 x = 1;
    required int32 y = 2;
    optional GhostState state = 4;
  }
  required AgentState pacman = 1;
  required AgentState red_ghost = 2;
  required AgentState pink_ghost = 3;
  required AgentState orange_ghost = 4;
  required AgentState blue_ghost = 5;
  required GameMode mode = 6;
  required int32 score = 8;
  required int32 lives = 11;

  enum GameMode {
    RUNNING = 0;
    PAUSED = 1;
  }

  enum GhostState {
    NORMAL = 0;
    FRIGHTENED = 1;
  }

}
